```
======================================================================
ERROR: test_get_apps_in_domain (corehq.apps.app_manager.tests.test_dbaccessors.DBAccessorsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_dbaccessors.py", line 86, in test_get_apps_in_domain
    apps = get_apps_in_domain(self.domain)
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/dbaccessors.py", line 148, in get_apps_in_domain
    apps = [get_correct_app_class(doc).wrap(doc) for doc in docs]
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/util.py", line 485, in get_correct_app_class
    raise DocTypeError()
DocTypeError
```
```
======================================================================
FAIL: test_instantiate_all (testapps.test_pillowtop.tests.test_settings.PillowtopSettingsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/testapps/test_pillowtop/tests/test_settings.py", line 35, in test_instantiate_all
    self.assertEqual(len(all_pillow_configs), len(expected_meta))
AssertionError: 52 != 51
```

--- a/corehq/apps/change_feed/document_types.py
+++ b/corehq/apps/change_feed/document_types.py
@@ -1,4 +1,6 @@
 from collections import namedtuple
+
+from corehq.apps.app_manager.util import app_doc_types
 from corehq.apps.change_feed.exceptions import MissingMetaInformationError
 from couchforms.models import all_known_formlike_doc_types
 from dimagi.utils.couch.undo import DELETED_SUFFIX
@@ -11,6 +13,7 @@ META = 'meta'
 COMMCARE_USER = 'commcare-user'
 WEB_USER = 'web-user'
 GROUP = 'group'
+APP = 'app'
 
 
 DocumentMetadata = namedtuple(
@@ -38,6 +41,8 @@ def _get_primary_type(raw_doc_type):
         return WEB_USER
     elif raw_doc_type in ('Group', 'Group-Deleted'):
         return GROUP
+    elif raw_doc_type in app_doc_types().keys():
+        return APP
     else:
         # at some point we may want to make this more granular
         return META

--- a/corehq/apps/change_feed/pillow.py
+++ b/corehq/apps/change_feed/pillow.py
@@ -60,6 +60,11 @@ def get_domain_db_kafka_pillow(pillow_id):
     return get_change_feed_pillow_for_db(pillow_id, couch_config.get_db_for_class(Domain))
 
 
+def get_application_db_kafka_pillow(pillow_id):
+    from corehq.apps.app_manager.models import Application
+    return get_change_feed_pillow_for_db(pillow_id, couch_config.get_db_for_class(Application))
+
+
 def get_change_feed_pillow_for_db(pillow_id, couch_db):
     kafka_client = get_kafka_client_or_none()
     processor = KafkaProcessor(

--- a/corehq/apps/change_feed/topics.py
+++ b/corehq/apps/change_feed/topics.py
@@ -1,10 +1,11 @@
-from .document_types import CASE, FORM, DOMAIN, META
+from .document_types import CASE, FORM, DOMAIN, META, APP
 
 # this is redundant but helps avoid import warnings until nothing references these
 CASE = CASE
 FORM = FORM
 DOMAIN = DOMAIN
 META = META
+APP = APP
 
 # new models
 CASE_SQL = 'case-sql'
@@ -28,6 +29,7 @@ ALL = (
     META,
     SMS,
     WEB_USER,
+    APP,
 )
 
 

--- a/corehq/apps/users/tests/test_get_domain_language.py
+++ b/corehq/apps/users/tests/test_get_domain_language.py
@@ -1,10 +1,12 @@
 from django.test import TestCase
 from elasticsearch.exceptions import ConnectionError
-from pillowtop.es_utils import completely_initialize_pillow_index
+
+from corehq.elastic import get_es_new, send_to_elasticsearch
+from corehq.pillows.mappings.app_mapping import APP_INDEX_INFO
+from pillowtop.es_utils import initialize_index_and_mapping
 
 from corehq.apps.app_manager.const import APP_V2
 from corehq.apps.app_manager.models import Application
-from corehq.pillows.application import AppPillow
 from corehq.util.elastic import delete_es_index
 from corehq.util.test_utils import trap_extra_setup
 
@@ -17,26 +19,26 @@ class TestDomainLanguages(TestCase):
     def setUpClass(cls):
         cls.domain = 'test-languages'
 
-        cls.pillow = AppPillow(online=False)
         with trap_extra_setup(ConnectionError):
-            completely_initialize_pillow_index(cls.pillow)
+            cls.es = get_es_new()
+            initialize_index_and_mapping(cls.es, APP_INDEX_INFO)
 
         cls.app1 = Application.new_app(cls.domain, 'My Application 1', APP_V2)
         cls.app1.langs = ['en', 'es']
         cls.app1.save()
-        cls.pillow.send_robust(cls.app1.to_json())
+        send_to_elasticsearch('apps', cls.app1.to_json())
         cls.app2 = Application.new_app(cls.domain, 'My Application 2', APP_V2)
         cls.app2.langs = ['fr']
         cls.app2.save()
-        cls.pillow.send_robust(cls.app2.to_json())
+        send_to_elasticsearch('apps', cls.app2.to_json())
 
-        cls.pillow.get_es_new().indices.refresh(cls.pillow.es_index)
+        cls.es.indices.refresh(APP_INDEX_INFO.index)
 
     @classmethod
     def tearDownClass(cls):
         cls.app1.delete()
         cls.app2.delete()
-        delete_es_index(cls.pillow.es_index)
+        delete_es_index(APP_INDEX_INFO.index)
 
     def test_get_domain_languages(self):
         self.assertEqual(

--- a/settings.py
+++ b/settings.py
@@ -1449,6 +1449,11 @@ PILLOWTOPS = {
         'corehq.pillows.reportcase.ReportCasePillow',
         'corehq.pillows.reportxform.ReportXFormPillow',
         {
+            'name': 'AppDbChangeFeedPillow',
+            'class': 'pillowtop.pillow.interface.ConstructedPillow',
+            'instance': 'corehq.apps.change_feed.pillow.get_application_db_kafka_pillow',
+        },
+        {
             'name': 'DefaultChangeFeedPillow',
             'class': 'pillowtop.pillow.interface.ConstructedPillow',
             'instance': 'corehq.apps.change_feed.pillow.get_default_couch_db_change_feed_pillow',

--- a/testapps/test_pillowtop/tests/test_app_pillow.py
+++ b/testapps/test_pillowtop/tests/test_app_pillow.py
@@ -1,0 +1,58 @@
+from django.test import TestCase
+from elasticsearch.exceptions import ConnectionError
+
+from corehq.apps.app_manager.models import Application
+from corehq.apps.app_manager.tests import AppFactory
+from corehq.apps.es import AppES
+from corehq.elastic import get_es_new
+from corehq.form_processor.tests.utils import FormProcessorTestUtils
+from corehq.pillows.application import AppPillow
+from corehq.pillows.mappings.app_mapping import APP_INDEX_INFO
+from corehq.util.elastic import ensure_index_deleted
+from corehq.util.test_utils import trap_extra_setup
+from pillowtop.es_utils import initialize_index_and_mapping
+from pillowtop.feed.couch import get_current_seq
+
+
+class AppPillowTest(TestCase):
+
+    domain = 'app-pillowtest-domain'
+
+    def setUp(self):
+        super(AppPillowTest, self).setUp()
+        FormProcessorTestUtils.delete_all_cases()
+        with trap_extra_setup(ConnectionError):
+            self.es = get_es_new()
+
+        ensure_index_deleted(APP_INDEX_INFO.index)
+        initialize_index_and_mapping(self.es, APP_INDEX_INFO)
+
+    def tearDown(self):
+        # ensure_index_deleted(APP_INDEX_INFO.index)
+        super(AppPillowTest, self).tearDown()
+
+    def test_app_pillow_couch(self):
+        since = get_current_seq(Application.get_db())
+
+        name = 'test app 1'
+        app = self._create_app(name)
+        pillow = AppPillow()
+        pillow.process_changes(since, forever=False)
+        self.es.indices.refresh(APP_INDEX_INFO.index)
+
+        # verify there
+        results = AppES().run()
+        self.assertEqual(1, results.total)
+        app_doc = results.hits[0]
+        self.assertEqual(self.domain, app_doc['domain'])
+        self.assertEqual(app['_id'], app_doc['_id'])
+        self.assertEqual(name, app_doc['name'])
+
+    def _create_app(self, name):
+        factory = AppFactory(domain=self.domain, name=name, build_version='2.11')
+        module1, form1 = factory.new_basic_module('open_case', 'house')
+        factory.form_opens_case(form1)
+        app = factory.app
+        app.save()
+        self.addCleanup(app.delete)
+        return app

